otherlibs/stdune/src/path.ml

```diff
@@ -1204,15 +1204,19 @@ let explode_exn t =
   | None -> Code_error.raise "Path.explode_exn" [ "path", to_dyn t ]
 ;;
 
-let relative_to_source_in_build_or_external ?error_loc ~dir s =
+let rec relative_to_source_in_build_or_external ?error_loc ~dir s =
   match Build.extract_build_context dir with
   | None -> relative ?error_loc (In_build_dir dir) s
   | Some (bctxt, source) ->
     let path = relative ?error_loc (In_source_tree source) s in
     (match path with
      | In_source_tree s ->
        In_build_dir (Build.relative (Build.of_string bctxt) (Source0.to_string s))
-     | In_build_dir _ | External _ -> path)
+     | In_build_dir _ -> path
+     | External _ -> (if (String.starts_with ~prefix:(External.to_string (External.cwd ())) s)
+                    then relative_to_source_in_build_or_external ?error_loc:error_loc ~dir (String.sub s ~pos:((String.length (External.to_string (External.cwd ())))+1) ~len:((String.length s)-(String.length (External.to_string (External.cwd ())))-1) )
+                    else path)
```

Comment:
`?error_loc:error_loc` is `?error_loc` I believe.

Putting `(External.to_string (External.cwd ()))` in a value would help with readability